pyqir-jit/src/jit.rs

```diff
@@ -24,56 +28,187 @@ pub fn run_module_file<P: AsRef<Path>>(path: P) -> Result<SemanticModel, String>
         .to_owned();
     let context_type = ContextType::File(&path_str);
     let context = BareContext::new(&ctx, context_type)?;
-    let model = run_module(&context.module)?;
-    Ok(model)
+    run_module(&context.module, entry_point)
 }
 
-pub fn run_module<'ctx>(module: &Module<'ctx>) -> Result<SemanticModel, String> {
+pub fn run_module<'ctx>(
+    module: &Module<'ctx>,
+    entry_point: Option<&str>,
+) -> Result<SemanticModel, String> {
     Target::initialize_native(&InitializationConfig::default()).unwrap();
-
     let default_triple = TargetMachine::get_default_triple();
 
     let target = Target::from_triple(&default_triple).expect("Unable to create target machine");
-
     assert!(target.has_asm_backend());
     assert!(target.has_target_machine());
 
     run_basic_passes_on(&module);
+    let entry_point = choose_entry_point(module_functions(module), entry_point)?;
 
     unsafe {
         BasicRuntimeDriver::initialize_qir_context(true);
         let _ = microsoft_quantum_qir_runtime_sys::foundation::QSharpFoundation::new();
-
         let _ = inkwell::support::load_library_permanently("");
         let execution_engine = module
             .create_jit_execution_engine(OptimizationLevel::None)
             .expect("Could not create JIT Engine");
         let simulator = Simulator::new(&module, &execution_engine);
         let main = execution_engine
-            .get_function::<unsafe extern "C" fn() -> ()>("QuantumApplication__Run")
+            .get_function::<unsafe extern "C" fn() -> ()>(&entry_point)
```

Comment:
@samarsha Is this still a safe type?

Reply:
I added a check before running the entry point to make sure it has no parameters and a void return type.